addon/ExperimentProvider.js

```diff
@@ -69,8 +110,15 @@ exports.ExperimentProvider = class ExperimentProvider {
   }
 
   destroy() {
-    simplePrefs.removeListener(OVERRIDE_PREF, this._onPrefChange);
-    this._data = {};
     this._experimentId = null;
+    Object.keys(this._experiments).forEach(experimentName => {
+      this._target.removeListener(experimentName, this._onPrefChange);
+    });
+  }
+
+  clearPrefs() {
```

Comment:
if for testability, maybe just make it an internal method and call it in `destroy`